NorthstarDedicatedTest/serverauthentication.cpp

```diff
@@ -141,6 +144,19 @@ void ServerAuthenticationManager::StopPlayerAuthServer()
 	m_playerAuthServer.stop();
 }
 
+char* ServerAuthenticationManager::VerifyPlayerName(void* player, char* authToken, char* name)
+{
+	AuthData authData = m_authData[authToken];
+
+	bool nameAccepted = (!*authData.username || name == "" || !strcmp(name, authData.username));
```

Comment:
Uhhh... `name == ""` is incorrect in C/C++ for char pointers (it means you're checking if name points to the const string "", which will essentially never be true). You want to do `!*name` instead.

Reply:
oh i just realised name shouldn't be checked here
should !*authData.username cover when it's missing and ""?

> Uhhh... `name == ""` is incorrect in C/C++ for char pointers (it means you're checking if name points to the const string "", which will essentially never be true). You want to do `!*name` instead.

Reply:
>  should !*authData.username cover when it's missing and ""?

Yes, since `authData.username` is a fixed-size char buffer and strncpy will always append the null terminator unless the source is larger than the destination (in which case you'll still have it anyways since you manually set the last byte to null).